### PR TITLE
build: add tool dependencies

### PR DESCRIPTION
**Pull request by mzgoddard**
Add extension build process, testing process, and documentation build process.

Add npm scripts:
- **build**: create an unpacked extension at `build/audion` and zip of that directory at `build/audion.zip`
  - bundle javascript
  - copy files
  - create an extension manifest
  - zip the extension package
- **test**:
  - lint all source files
  - create documentation
  - run unit tests and collect coverage
  - run integration tests on with the unpacked extension

Add tool dependencies with npm for the build process:
- **webpack**: bundles the javascript
- **mustache**: used to create the manifest
- **yazl**: used to zip the extension (yazl is the zip library used in gulp-zip)
- **eslint**, **eslint-config-google**: enforce code style
- **prettier**: auto-format code
- **husky**, **lint-staged**: lint staged files during a pre-commit hook
- **jest**: run unit and integration tests
- **babel-jest**: transform es module syntax to commonjs
- **puppeteer**, **jest-puppeteer**: launch chrome with the unpacked extension for integration tests
- **jsdoc**: create documentation from code comments
- **@pixi/jsdoc-template**: a jsdoc template with a search field

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,60 @@
+{
+  "name": "audion",
+  "version": "1.0.0",
```
Comment (hoch) [inline package.json:3]: nit: Perhaps 3.0.0?

File: package.json [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  "description": "Traces web audio API calls.",
```
Comment (hoch) [inline package.json:4]: nit: A Chrome DevTools extension traces Web Audio API calls and visualizes in the DevTools.

File: package.json [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  "main": "index.js",
+  "engines": {
+    "node": "14"
+  },
+  "devDependencies": {
+    "@babel/core": "^7.14.6",
```
Comment (hoch) [inline package.json:10]: Why do we need Babel? I thought we'd use vanilla JS for the implementation.

File: package.json [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    "@babel/plugin-transform-modules-commonjs": "^7.14.5",
+    "@pixi/jsdoc-template": "^2.6.0",
+    "babel-jest": "^27.0.6",
```
Comment (hoch) [inline package.json:13]: I guess the Jest test framework requires Babel?

File: package.json [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    "eslint": "^7.30.0",
+    "eslint-config-google": "^0.14.0",
+    "husky": ">=6",
+    "jest": "^27.0.6",
+    "jest-puppeteer": "^5.0.4",
+    "jsdoc": "^3.6.7",
+    "lint-staged": ">=10",
+    "mustache": "^4.2.0",
```
Comment (hoch) [inline package.json:21]: Why do we need an additional template library?

File: package.json [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    "test:unit": "jest --config src/.jest.config.json",
+    "test": "npm run test:lint && npm run test:jsdoc && npm run test:unit && npm run test:integration"
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/GoogleChrome/audion.git"
+  },
+  "keywords": [],
+  "author": "",
+  "license": "ISC",
```
Comment (hoch) [inline package.json:51]: Not sure if this is the correct license for projects in the GoogleChrome org. See this for an example:
https://github.com/GoogleChrome/omnitone/blob/master/LICENSE

**Review by hoch — COMMENTED** [review]
Looks great. I have some questions on some details, but I could understand the general direction of this PR.

File: package.json (hunk 1 above)
Comment (mzgoddard) [reply package.json:3]: Sounds good to me.

File: package.json (hunk 3 above)
Comment (mzgoddard) [reply package.json:10]: It is used in the `jest` configuration. `jest` doesn't support es module syntax out of the box. The babel configuration here uses one babel plugin `@babel/plugin-transform-modules-commonjs` to transform import/export statements to commonjs syntax.

File: package.json (hunk 6 above)
Comment (mzgoddard) [reply package.json:51]: I'll change that. ISC is the `npm init` default and I saw that audion's main branch package.json license field is set to the same. https://github.com/GoogleChrome/audion/blob/main/package.json#L10 I didn't look at the LICENSE file though, which matches the repo you linked. https://github.com/GoogleChrome/audion/blob/main/LICENSE

File: package.json (hunk 5 above)
Comment (mzgoddard) [reply package.json:21]: Additional? I think it's the only template library depended on currently.

Comment (hoch) [reply package.json:21]: Mustache might have additional values, but I thought JS template literal can do many things that Mustache does. I am just curious, and this is not a blocker.

File: package.json (hunk 4 above)
Comment (mzgoddard) [reply package.json:13]: It's one option. `babel-jest` provides the means to use babel on files jest loads. Another option would be to use node's module support but I think that would be more trouble.

**Comment by mzgoddard** [comment]
Pushed with my bocoup email to comply with google-cla bot. Will push in a bit with a few changes to implement comment responses.

File: package.json (hunk 5 above)
Comment (mzgoddard) [reply package.json:21]: Kk. I'll revisit mustache.

**Pull request merged**